### PR TITLE
Adding the capability to send index data to hydra

### PR DESCRIPTION
**Pull request by tahmidefaz**
I am yet to receive full access to the service account to be able to send over data to the search service, but from the limited access that I got + the documentation, I think this will work even though I wasn't able to properly test it.

The script successfully fetched the JWT token, but only gets rejected by Hydra because it's missing some auth keys in it.

### CODE REVIEW

File: packages/sitemap/src/collector.ts [hunk 1]
```diff
@@ -52,3 +53,47 @@ export const collector = async (config: ReadonlyArray<Readonly<APIConfiguration>
...
+
+    if (syncActive == "true") {
+        const jwtFetchUrl = process.env.JWT_FETCH_URL as string;
+        const clientId = process.env.CLIENT_ID;
+        const clientSecret = process.env.CLIENT_SECRET;
+
+        const authHeader = `Basic ${Buffer.from(`${clientId}:${clientSecret}`).toString('base64')}`;
+        const jwtRequestBody = 'grant_type=client_credentials';
+
+        axios.post(jwtFetchUrl, jwtRequestBody, {
```
Comment (josejulio) [inline packages/sitemap/src/collector.ts:68]: should return this to be able to await for it outside

**Comment by tahmidefaz** [comment]
@josejulio just updated with await. Is that what you had in mind?

**Comment by josejulio** [comment]
> @josejulio just updated with await. Is that what you had in mind?

yes, thanks!

**Comment by tahmidefaz** [comment]
@josejulio This doesn't seem to be working on pre-prod environments for some reason. But, when I try to do similar things using `curl`, it works just fine. I am guessing it might be related to some proxy settings with `nodejs`. They have not enabled this for us in production yet, so, I was not able to test it there.

**Comment by josejulio** [comment]
> @josejulio This doesn't seem to be working in pre-prod environments for some reason. But, when I try to do similar things using `curl`, it works just fine. I am guessing it might be related to some proxy settings with `nodejs`. They have not enabled this for us in production yet, so, I was not able to test it there.
> 
> I used the following commands to set up proxy:

should we merge this and try on gitlab?

**Comment by tahmidefaz** [comment]
Sounds good to me! I think their `QA` environment went down this afternoon, and it's not showing any signs of coming back up. `QA` is the only environment they have enabled this for us. :joy: 
Do you want to wait until Monday to do that? I also just finished setting up the env vars on gitlab.

**Comment by josejulio** [comment]
> Do you want to wait until Monday to do that? I also just finished setting up the env vars on gitlab.

Sounds good, lets do this on Monday.

**Review by josejulio — APPROVED** [review]
lgtm

**Pull request merged**